### ISSUE TITLE
How to execute scripts on test network?

### ISSUE THREAD

**Issue by frods** [JoinColony/colonyJS#243]
Hello, I have written a script to create a colony. https://github.com/Pioneercoinio/PioneerColony

It is tested to be working with Truffle, so I feel confident that it works. I would like to run it against Rinkeby to further validate it and to have the colony there so others can interact with it.

I have tried to do this with the node-metamask library. Using this I have issues with the EthersAdaptor being able to handle requests.
https://github.com/Pioneercoinio/PioneerColony/blob/MetaMaskClient/src/metamask_client.js

I love the idea of using metamask because it is easy for people to use. My next aim is to build a simple site and run the script in the browser. But since I would still be using web3 I will likely run into the same issues with EthersAdaptor.

Any links to examples or guidance would be greatly appreciated.

**Comment by gichiba** [comment]
Hey @frods, this is great! Web3 support is still incomplete as of writing, I think, but I think @ryanchristo might be able to help out; he's finishing off a big upgrade to our hackathonStarter kit, which contains plenty of examples: 
https://github.com/JoinColony/hackathonStarter

**Comment by ryanchristo** [comment]
@frods Yes! Check out [hackathonStarter](https://github.com/JoinColony/hackathonStarter) when you get a chance (we are changing the name soon so if the previous link doesn't work at the time you are reading this, try [colonyStarter](https://github.com/JoinColony/colonyStarter)).

We don't have an example in the starter project using `node-metamask` yet but I will add it to the list. Let me tinker around and spin up an example for you. Hang tight!

**Comment by frods** [comment]
Thanks for the responses. I had reviewed the hackathonStarter kit before posting. Unless I'm missing something it only describes deploying to the trufflepig network.

My main question is about deploying to Rinkeby. If you do that through Truffle, how do you manage the wallet?

I'm happy with using whatever is the current standard. I have contributed to ```node-metamask``` to help me get farther but hit a wall with the EthersAdaptor.

It's great to hear that work is continuing on support for Web3. What is the status with that? Is there anything I can help with to get it done more quickly?

**Comment by ryanchristo** [comment]
Hey @frods! Check out [this branch](https://github.com/JoinColony/hackathonStarter/tree/u/frod/rinkeby-node-metamask) in the hackathonStarter repo (we are changing the name soon so if the previous link doesn't work at the time you are reading this, try [this link](https://github.com/JoinColony/colonyStarter/tree/u/frod/rinkeby-node-metamask)).

In this branch you will find a `connectRinkeby` example and a `connectMetamask` example. You will also notice that I am using a version of `node-metamask` not yet released. I will add the two examples to the master branch of the hackathonStarter project when I have a better idea of where they will fit in.

Let me know if this answers your questions so we can close the issue.

**Comment by frods** [comment]
Thanks for the looking up the network was a nice touch. otherwise it is what I am already doing.

This is the error I am hitting when trying to create a colony

```
action executed, request_id 73e04404-9ada-141a-4f6d-b30e2c4768a6, payload {"address":"0xd4c145ebdc7f072d10a07b8ea4515af996ee437c","topics":["0x6747f57524d6303c6e3d698d8b6ab540620b0455ccde2e5aa047d8a179787d74","0x000000000000000000000000000000000000000000000000000000000000004d","0x000000000000000000000000cf2f453c8c06e0b645a8a2c69c769019f2be9515"],"data":"0x","blockNumber":2795882,"transactionHash":"0xbe8d67e198127e0312d34a056caa45a4e9ea29594ac62dd420212f6805a593e2","transactionIndex":5,"blockHash":"0x63bc0bbb9457a30d7c8e85f5dee368aff3bdfd158462eb94d1e2d2089348b0c8","logIndex":9,"removed":false}
(node:224) UnhandledPromiseRejectionWarning: Error: not an array
    at /Users/philip/github/pioneercoinio/PioneerColony/node_modules/ethers/providers/provider.js:78:44
    at /Users/philip/github/pioneercoinio/PioneerColony/node_modules/ethers/providers/provider.js:881:37
```
I've updated my branch with the latest state
https://github.com/Pioneercoinio/PioneerColony/tree/MetaMaskClient

You should be able to reproduce what I'm seeing with
```yarn install```
```yarn metamask```
then follow the prompts

**Comment by ryanchristo** [comment]
I was able to run your code on `rinkeby` and [verify the completed transaction on Etherscan](https://rinkeby.etherscan.io/address/0xd4c145ebdc7f072d10a07b8ea4515af996ee437c) but I am running into the same `UnhandledPromiseRejectionWarning`, which is being thrown from the installed `ethers` package. It looks like there is an issue with the values in the payload from `getTransaction`, which are being read as strings and not getting converted to hex values. I'll keep investigating.

I was able to successfully run the method and verify the completed transaction with Metamask using a local test network. If you want to try it out yourself, I added the option to the `connectMetamask` example in [the same branch](https://github.com/JoinColony/hackathonStarter/tree/u/frod/rinkeby-node-metamask) (you will need to start Ganache, deploy the contracts, start Trufflepig, and set your Metamask network to `Localhost 8545`).

**Comment by ryanchristo** [comment]
Hey @frods!

The examples have been added to the `colony-starter-basic` package in [colonyStarter](https://github.com/JoinColony/colonyStarter). I also resolved the error being thrown from the `ethers` package with a `formatResult` method in [node-metamask](https://github.com/JoinColony/node-metamask).

I tested your code again with the latest version of `node-metamask` (version `1.1.1`) and I was able to successfully execute the code both on Rinkeby and when connected to a local test network.

I am going to close this issue. If you have any issues in the future related to examples or if you are not sure how to do something, feel free to open an issue in [colonyStarter](https://github.com/JoinColony/colonyStarter).
